imaginairy/schema.py

```diff
@@ -172,385 +133,543 @@ def handle_b64(value: Any) -> "LazyLoadingImage":
+    #     return v
 
-        if v is None:
-            v = get_model_default_image_size(info.data["model"])
+    # @field_validator("fix_faces_fidelity", mode="before")
+    # def validate_fix_faces_fidelity(cls, v):
+    #     if v is None:
+    #         return 0.2
 
-        return v
+    #     return v
 
-    @field_validator("caption_text", mode="before")
-    def validate_caption_text(cls, v):
-        if v is None:
-            v = ""
+    # @field_validator("sampler_type", mode="after")
+    # def validate_sampler_type(cls, v, info: core_schema.FieldValidationInfo):
+    #     from imaginairy.samplers import SamplerName
 
-        return v
+    #     if v is None:
+    #         v = config.DEFAULT_SAMPLER
 
-    @property
-    def prompts(self):
-        return self.prompt
+    #     v = v.lower()
+
+    #     if info.data.get("model") == "SD-2.0-v" and v == SamplerName.PLMS:
+    #         raise ValueError("PLMS sampler is not supported for SD-2.0-v model.")
+
+    #     if info.data.get("model") == "edit" and v in (
+    #         SamplerName.PLMS,
+    #         SamplerName.DDIM,
+    #     ):
+    #         msg = "PLMS and DDIM samplers are not supported for pix2pix edit model."
+    #         raise ValueError(msg)
+    #     return v
+
+    # @field_validator("steps")
+    # def validate_steps(cls, v, info: core_schema.FieldValidationInfo):
+    #     from imaginairy.samplers import SAMPLER_LOOKUP
+
+    #     if v is None:
+    #         SamplerCls = SAMPLER_LOOKUP[info.data["sampler_type"]]
+    #         v = SamplerCls.default_steps
+
+    #     return int(v)
+
+    # @model_validator(mode="after")
+    # def validate_init_image_strength(self):
+    #     if self.init_image_strength is None:
+    #         if self.control_inputs:
+    #             self.init_image_strength = 0.0
+    #         elif self.outpaint or self.mask_image or self.mask_prompt:
+    #             self.init_image_strength = 0.0
+    #         else:
+    #             self.init_image_strength = 0.2
+
+    #     return self
+
+
+@dataclass
+class ImaginePrompt:
+    prompt: list[WeightedPrompt]
+    negative_prompt: list[WeightedPrompt]
+    prompt_strength: float
+    init_image: InitImageConfig | None
+    width: int
+    height: int
+    seed: int
+    steps: int
+    model: str
+    model_config: Path  # ?: what is this?
+    sampler_type: SamplerType
+    control_inputs: list[ControlNetInput]
+    outpaint: str  # TODO: implement OutpaintConfig class
+    face_fixer: FaceFixerConfig
+    tiling: TileMode
+    caption_text: str | None
+
+    def __init__(
+        self,
+        prompt: PromptInput | None = None,
+        negative_prompt: PromptInput | None = None,
+        prompt_strength: float = 7.5,
+        init_image: InitImageInput | None = None,
+        width: int | None = None,
+        height: int | None = None,
+        seed: int | None = None,
+        steps: int = 30,
```

Comment:
It needs to be default None since the appropriate amount depends a lot on the sampler and things should "just work" when a different sampler is selected.